docs/apm/troubleshooting.asciidoc

```diff
@@ -69,3 +69,31 @@ or because something is happening to the request that the Agent doesn't understa
 To resolve this, you'll need to head over to the relevant {apm-agents-ref}[Agent documentation].
 Specifically, view the Agent's supported technologies page.
 You can also use the Agent's public API to manually set a name for the transaction.
+
+==== Fields are not indexed or searchable
+
+In Elasticsearch, index templates are used to define settings and mappings that determine how fields should be analyzed.
+The recommended index template file for APM Server is installed by the APM Server packages.
+This template, by default, enables and disables indexing on certain fields.
+
+As an example, some agents store cookie values in `http.request.cookies`.
+Since `http.request` has disabled dynamic indexing, and `http.request.cookies` is not declared in a custom mapping,
+the values in `http.request.cookies` are not indexed and thus not searchable.
+
+There are two things you can do to if you'd like to ensure a field is indexed and searchable:
+
+1. Index your additional data as {apm-overview-ref}/metadata.html[labels].
+These are dynamic by default, which means they will be indexed and become searchable and aggregatable.
+
+2. Use the {apm-server-ref}/configuration-template.html[`append_fields`] feature. As an example,
+adding the following to `apm-server.yml` will enable dynamic indexing for `http.request.cookies`:
+
+[source,yml]
+----
+setup.template.enabled: true
+setup.template.overwrite: true
+setup.template.append_fields:
+  - name: http.request.cookies
+    type: object
+    dynamic: true
+----
```

Comment:
i wonder if it's also appropriate to point to the index pattern management UI `/app/kibana#/management/kibana/index_patterns` to check and see that `apm-*` or (whatever pattern the user uses for apm indices) exists there and their fields are all defined. If not, they can create an index pattern in that UI.

Reply:
Good idea. Added.